### PR TITLE
Feature/release process

### PR DESCRIPTION
**Pull request by MarcusJones**
Copied over squid RELEASE_PROCESS.md, and generalized to all python packages.

### CODE REVIEW

File: doc/development/python-release.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+This document describes the general way in which Python packages are released. 
+
+## Prerequistes: 
+
+- Optionally create a new virtual environment for the `bumpversion`  package, and other release tools
+- Install the bumpversion package , `pip install bumpversion`
+- Following git-flow process
+- Running Travis in the repo, i.e. `.travis.yml` exists and has the `deploy` section defined, (see below)
+- bumpversion.cfg exists, example below
```
Comment (ttmc) [inline doc/development/python-release.md]:
```suggestion
- `.bumpversion.cfg` exists, example below
```

File: doc/development/python-release.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+[bumpversion:file:squid_py/__init__.py]
+search = __version__ = '{current_version}'
+replace = __version__ = '{new_version}'
+
+[bumpversion:file:docs/source/conf.py]
+search = {current_version}
+replace = {new_version}
+```
+
```
Comment (ttmc) [inline doc/development/python-release.md]: The `__init__.py` section is only needed if there's a line in the `__init__.py` file that sets `__version__`

The `docs/source/conf.py` section is only needed if that file exists.

Generally speaking, bumpversion should update all files that list the version. You can find those files by doing a search across the repo for the current version number (e.g. 0.2.16).

File: doc/development/python-release.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Python Release Process
```
Comment (ttmc) [inline doc/development/python-release.md]: Maybe rename the file `python-release-process.md` just to be 100% clear.

File: doc/development/python-release.md (hunk 1 above)
Comment (ttmc) [reply doc/development/python-release.md]: Maybe also add that they shouldn't have any `bumpversion` stuff in `setup.cfg`: bumpversion looks in `.bumpversion.cfg` first and `setup.cfg` second but we should just standardize on always using `.bumpversion.cfg`.

File: doc/development/python-release.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  user: ${PYPI_USER}
+  password: ${PYPI_PASSWORD}
+  on:
+    tags: true
+    repo: oceanprotocol/squid-py
+    python: 3.6
+  ```
+
+- Go to Travis and check the Travis job. It should deploy a new release to PyPI.
+- Check PyPI for the new release
```
Comment (ttmc) [inline doc/development/python-release.md]: Maybe also check Docker Hub to see if it generated a new Docker image (if it should).

**Review by ttmc — COMMENTED** [review]
It might also be necessary to update the `.travis.yml` file because it may contain a string like `KEEPER_VERSION=v0.5.1`

File: doc/development/python-release.md (hunk 3 above)
Comment (MarcusJones) [reply doc/development/python-release.md]: done

File: doc/development/python-release.md (hunk 4 above)
Comment (MarcusJones) [reply doc/development/python-release.md]: I think we should keep the scope here to only PyPI, seperate doc for Docker.

**Review by eruizgar91 — APPROVED** [review]
Great job!

**Pull request merged**